be/src/exec/vectorized/connector_scan_node.cpp

```diff
@@ -161,17 +161,21 @@ Status ConnectorScanNode::open(RuntimeState* state) {
     SCOPED_TIMER(_runtime_profile->total_time_counter());
     RETURN_IF_ERROR(ScanNode::open(state));
     RETURN_IF_ERROR(_data_source_provider->open(state));
+
     return Status::OK();
 }
 
 Status ConnectorScanNode::_start_scan_thread(RuntimeState* state) {
     for (TScanRangeParams& scan_range : _scan_ranges) {
         _create_and_init_scanner(state, scan_range.scan_range);
     }
+    _num_scanners = _pending_scanners.size();
```

Comment:
when will `_num_scanners == 0` happen?

Reply:
@DorianZheng when there is no scan file in this table.

Reply:
OK, got it. Why not prevent this case on FE?

Reply:
Yes. That would also solve the problem. It generates an EMPTY NODE.

Reply:
Why does it generate an empty node instead of return early?